src/zkevm_specs/evm/execution/mul_div_mod.py

```diff
@@ -0,0 +1,71 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ...util import FQ
+
+
+def mul_div_mod(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    # The opcode value for MUL, DIV and MOD is 2, 4, 6. When the opcode is MUL,
+    # (Opcode.DIV - opcode) * (Opcode.MOD - opcode) is 8. To make `is_mul` be
+    # either 0 or 1, we need to divide the product by 8, which is equivalent to
+    # multiply it by inversion of 8. Similarly, we also need to multiply the
+    # inversion of 4 and 8 for `is_div` and `is_mod` respectively.
+    is_mul = (Opcode.DIV - opcode) * (Opcode.MOD - opcode) * FQ(8).inv()
+    is_div = (opcode - Opcode.MUL) * (Opcode.MOD - opcode) * FQ(4).inv()
+    is_mod = (opcode - Opcode.MUL) * (opcode - Opcode.DIV) * FQ(8).inv()
+
+    pop1 = instruction.stack_pop()
+    pop2 = instruction.stack_pop()
+    push = instruction.stack_push()
+
+    # this part corresponds to witness assignment in the zkevm circuit
+    if is_mul == 1:
+        a = pop1
+        b = pop2
+        c = instruction.rlc_encode(0, 32)
```

Comment:
Do we have to rlc encode 0 when it's already 0?

Reply:
It's to keep the type consistent with other cases.